crates/sui-core/src/authority/authority_store.rs

```diff
@@ -381,6 +384,18 @@ impl AuthorityStore {
                 }
             }
             .ok_or_else(|| SuiError::from(kind.object_not_found_error()))?;
+            if !obj.is_immutable() {
```

Comment:
I still think if we have this, that we should put this in the input checker, feels strange here. 

But I feel like this limit is unnecessary? We already have a limit on the number of inputs. Do we really need a secondary limit here? (and at the current value I think it will be impossible to hit)

Reply:
> I still think if we have this, that we should put this in the input checker, feels strange here.

This is hit in the input checker. The name of the function is `check_input_objects`, which is called by the `check_transaction_input`

> But I feel like this limit is unnecessary? We already have a limit on the number of inputs. Do we really need a secondary limit here? (and at the current value I think it will be impossible to hit)

Mutable inputs contribute a lot to effects size, that was the main motivation for having a separate knob. But I can remove it or make it high enough that we wont hit it right now. These numbers can always be changed.

Reply:
I just checked, we actually don't have a limit on inputs right now! I think we should probably do that first before doing some smaller, separate knob

Reply:
Done, good callout.